src/contract_spec.ts

```diff
@@ -687,11 +765,322 @@ function findCase(name: string) {
   };
 }
 
-function readObj(args: object, input: xdr.ScSpecFunctionInputV0): any {
-  let inputName = input.name().toString();
-  let entry = Object.entries(args).find(([name, _]) => name === inputName);
-  if (!entry) {
-    throw new Error(`Missing field ${inputName}`);
+const PRIMITIVE_DEFINITONS = {
+  U64: {
+    type: 'integer',
+    minimum: 0,
+    maximum: 18446744073709551615
```

Comment:
This value is precision truncated, it evaluates to this in JS because numbers in JS can only store all integers up to 53-bits, at which point precision is lost:
```
18446744073709552000
```

Reply:
Yeah I was wondering if we should use strings to ensure the precision.  It would be great if JSON had support for bigint (which it does on paper since there is no size limit on numbers).

Reply:
In Stellar JSON APIs we usually use strings for anything over 32bit. Although we haven't done that in the Rust XDR JSON (https://github.com/stellar/xdrgen/issues/181).

The problem is JS not providing anyway to parse numbers into BigInts 😢.